shell_util/remote_connection.py

```diff
@@ -45,3 +45,10 @@ def __new__(cls, *args, **kwargs):
         obj.ssh_connect_with_retries(server.ip, server.ssh_username,
                                      server.ssh_password, server.ssh_key)
         return obj
+
+    @staticmethod
+    def delete_info_for_server(server, ipaddr=None):
+        ipaddr = ipaddr if ipaddr else server.ip
+        if ipaddr in RemoteMachineShellConnection.__info_dict:
+            del RemoteMachineShellConnection.__info_dict[ipaddr]
```

Comment:
Should we also close the connection here ?

Reply:
We don't store the connection in this particular class. All we are storing is the info dictionary to fetch the right class.